app/models/volunteer.rb

```diff
@@ -124,66 +129,77 @@ def has_main_region?
   end
 
   def main_region
-    self.regions[0]
+    regions[0]
   end
 
   def region_ids
-    self.regions.collect{ |r| r.id }
+    regions.collect(&:id)
   end
 
-  def admin_region_ids(strict=false)
-    admin_regions(strict).collect { |r| r.id }
+  def admin_region_ids(strict = false)
+    admin_regions(strict).collect(&:id)
   end
 
-  def admin_regions(strict=false)
-    if self.super_admin? and not strict
+  def admin_regions(strict = false)
+    if super_admin? && !strict
       Region.all
     else
-      self.assignments.collect{ |a| a.admin ? a.region : nil }.compact
+      assignments.collect { |a| a.admin ? a.region : nil }.compact
     end
   end
 
-  def in_region? region_id
-    self.region_ids.include? region_id
-  end
-
-  # better first-time experience: if there is only one region, add the user to that one automatically when they sign up
-  def auto_assign_region
-    if Region.count==1 and self.regions.count==0
-      Assignment.add_volunteer_to_region self, Region.first
-      logger.info "Automatically assigned new user to region #{self.regions.first.name}"
-    end
+  def in_region?(region_id)
+    region_ids.include? region_id
   end
 
   def current_absences
-    self.absences.keep_if{ |a| a.start_date < Date.today and a.stop_date > Date.today }
+    today = Date.today
+
+    absences.keep_if do |a|
+      a.start_date < today && a.stop_date > today
```

Comment:
Can this be pushed off onto the DB, instead of doing this in Ruby? `absences.where('start_date < ? AND stop_date > ?, today, today)`